lib/upload_creator.rb

```diff
@@ -427,4 +427,30 @@ def add_metadata!
     @upload.secure = UploadSecurity.new(@upload, @opts).should_be_secure?
   end
 
+  private
+
+  def animated?
+    return @animated if @animated != nil
+
+    @animated ||= begin
+      is_animated = FastImage.animated?(@file)
+      type = @image_info.type.to_s
+
+      if is_animated != nil
+        # FastImage will return nil if it cannot determine if animated
+        is_animated
+      elsif type != "gif" && type != "webp"
```

Comment:
I don't understand this `if`.

I want to use this method to detect animated `webp` but it's running only when type is different from `webp` ?

Reply:
Yes, sorry, my mistake. I did not rewrite the condition when I refactored the early return.

Reply:
Maybe you meant to test only if it's different from `gif`? Since that is supported by FastImage anyway.

Reply:
I added `gif` just as a failsafe.